File: DB_lez07_dipendenza_ciclica/DB_lez07_dipendenza_ciclica/Models/Albergo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DB_lez07_dipendenza_ciclica.Models
{
    internal class Albergo
    {
        public string Nome { get; set; } = null!;
        public string Indirizzo { get; set; } = null!;
        public int Stelle { get; set; } = 1;

        public List<Camera>? Camere { get; set; } = null;

        public void AggiungiCamera(Camera cam)
        {
            if(Camere is null)
                Camere = new List<Camera>();

            Camere.Add(cam);

            Console.WriteLine("Camera aggiunta con successo");
        }

        public string StampaDettaglio()
        {
            return $"[ALBERGO] {Nome} {Indirizzo} {Stelle}";
        }

        public override string ToString()
        {
            string camereString = "";
            if(Camere is not null)
                foreach(Camera cam in Camere)
                {
                    camereString += cam.StringaDettaglio();
                }

            return $"[ALBERGO] {Nome} {Indirizzo} {Stelle} Elenco: [{camereString}]";
        }
    }
}

File: Lez_TS/lez10_composizione/Albergo.ts
import Camera from "./Camera";

export default class Albergo{
    private indirizzo: string | undefined;
    private elenco: Camera[] | undefined;

    constructor(indirizzo: string){
        this.indirizzo = indirizzo;
    }

    aggiungiCamera(cam: Camera) : void{
        if(!this.elenco)
            this.elenco = new Array()

        this.elenco.push(cam);
    }

    stampaAlbergo(): void {
        console.log(`Albergo: ${this.indirizzo}`);

        if(this.elenco)
            this.elenco.forEach(element => {
                element.dettaglioCamera();
            });
    }
}